optaplanner-examples/src/main/java/org/optaplanner/examples/travelingtournament/domain/TravelingTournament.java

```diff
@@ -114,12 +114,8 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        HashCodeBuilder hashCodeBuilder = new HashCodeBuilder();
-        for (Match match : matchList) {
-            // Notice: we don't use hashCode()
-            hashCodeBuilder.append(match.solutionHashCode());
-        }
-        return hashCodeBuilder.toHashCode();
+        int[] hashes = matchList.stream().mapToInt(Match::solutionHashCode).toArray();
```

Comment:
"Notice: we don't use hashCode()" - why not? 
Ow... On Match.java
```
    /**
     * The normal methods {@link #equals(Object)} and {@link #hashCode()} cannot be used
     * because the rule engine already requires them (for performance in their original state).
     * @see #solutionHashCode()
     */
    public boolean solutionEquals(Object o) {
```
Ok, that didn't age well... (the code was written in 2007 or even 2006, because it's part of the first commit that came from SourceForge)
That javadocs is probably not true any more. Drools doesn't use Java equality since Drools 4 or so.
I've created a jira: https://issues.redhat.com/browse/PLANNER-1746